src/NuGet.Core/NuGet.Commands/SignCommand/CertificateProvider.cs

```diff
@@ -122,7 +128,12 @@ private static async Task<X509Certificate2> LoadCertificateFromFileAsync(Certifi
 
             if (!string.IsNullOrEmpty(options.CertificatePassword))
             {
-                cert = new X509Certificate2(options.CertificatePath, options.CertificatePassword); // use the password if the user provided it.
+                // use the password if the user provided it
+#if NET9_0_OR_GREATER
+                cert = X509CertificateLoader.LoadPkcs12FromFile(options.CertificatePath, options.CertificatePassword);
+#else
+                cert = new X509Certificate2(options.CertificatePath, options.CertificatePassword);
```

Comment:
The preferred fix is to add a package reference to Microsoft.Bcl.Cryptography and always call X509CertificateLoader; but we don't have the power to mark things Obsolete in older TFMs.

If adding the package is unacceptable, then this works.  The main concern is "don't get surprised by a PFX", but this expects a PFX.  There are also some subtle differences in how PFX loading works on Windows between the two, mainly fixing issues with loading very old files or loading the same file in parallel (making Microsoft.Bcl.Cryptography/X509CertificateLoader the better choice)

Reply:
Is `X509CertificateLoader` brand new, not even in .NET 9 Preview 6?  I had a quick look at the preview 6 package on nuget.org, but it doesn't appear to have the API.

For this reason, plus the fact that NuGet doesn't use the Arcade SDK, means I don't think NuGet can use this package for .NET 9.0.100 SDK.  Maybe after .NET 9 goes GA. For now, I think we need to take this conditional compilation change.